### PR TITLE
Fixes #1341: AudioBuffer.updateData only allows direct buffers

### LINKED ISSUE #1341: AudioBuffer.update accepts any type of ByteBuffer
There is no check whether the provided buffer is a nativeBuffer(as required), probalby not easy to implement, but a hint in the javadoc would imho already be a great improvement...

https://hub.jmonkeyengine.org/t/solved-playing-audio-from-audiobuffer-in-audionode-causes-jre-to-die/43091/9

### PR DESCRIPTION
**Pull request by plassma**
See forum topic: 
https://hub.jmonkeyengine.org/t/solved-playing-audio-from-audiobuffer-in-audionode-causes-jre-to-die/43091/12

### CODE REVIEW

File: jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java [hunk 1]
```diff
@@ -87,8 +87,12 @@ public String toString(){
     /**
      * Update the data in the buffer with new data.
      * @param data
+     * @throws IllegalArgumentException if the provided buffer is not a direct buffer
      */
     public void updateData(ByteBuffer data){
+        if (!data.isDirect())
```
Comment (MeFisto94) [inline jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java:93]: My personal opinion/style would want `{}s` even for one-line statements, but I don't know:
- What our jme desired standard would be and
- how our current code looks right now

Comment (pspeed42) [reply jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java:93]: Yeah, especially in this case where the line is pretty long.

Comment (Ali-RS) [reply jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java:93]: Yeah, `{}` is preferred.

**Comment by MeFisto94** [comment]
Also for future PRs having a commit message like `Fixes #1341 AudioBuffer.update: Only allow direct buffers` or something would be perfect :)

**Comment by stephengold** [comment]
It's not too late to change the commit message using 'push --force'!

File: jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java (hunk 1 above)
Comment (stephengold) [reply jme3-core/src/main/java/com/jme3/audio/AudioBuffer.java]: I got impatient and standardized the coding style.

**Comment by stephengold** [comment]
Integrated into the v3.3 branch at c85e2bc .

**Pull request merged**